src/store.rs

```diff
@@ -0,0 +1,172 @@
+use std::path::{Path, PathBuf};
+
+use rocksdb::{ColumnFamilyDescriptor, DBWithThreadMode, MultiThreaded};
+
+use crate::{block::store::BlockStore, state::ledger::store::LedgerStore};
+
+#[derive(Debug)]
+pub struct IndexerStore {
+    db_path: PathBuf,
+    database: DBWithThreadMode<MultiThreaded>,
+}
+
+impl IndexerStore {
+    pub fn new_read_only(path: &Path, secondary: &Path) -> anyhow::Result<Self> {
+        let database_opts = rocksdb::Options::default();
+        let database = rocksdb::DBWithThreadMode::open_cf_as_secondary(
+            &database_opts,
+            path,
+            secondary,
+            vec!["blocks", "ledgers"],
+        )?;
+        Ok(Self {
+            db_path: PathBuf::from(path),
+            database,
+        })
+    }
+    pub fn new(path: &Path) -> anyhow::Result<Self> {
+        let mut cf_opts = rocksdb::Options::default();
+        cf_opts.set_max_write_buffer_number(16);
+        let blocks = ColumnFamilyDescriptor::new("blocks", cf_opts.clone());
+        let ledgers = ColumnFamilyDescriptor::new("ledgers", cf_opts);
+
+        let mut database_opts = rocksdb::Options::default();
+        database_opts.create_missing_column_families(true);
+        database_opts.create_if_missing(true);
+        let database = rocksdb::DBWithThreadMode::open_cf_descriptors(
+            &database_opts,
+            path,
+            vec![blocks, ledgers],
+        )?;
+        Ok(Self {
+            db_path: PathBuf::from(path),
+            database,
+        })
+    }
+
+    pub fn db_path(&self) -> &Path {
+        &self.db_path
+    }
+}
+
+impl BlockStore for IndexerStore {
+    fn add_block(&self, block: &crate::block::precomputed::PrecomputedBlock) -> anyhow::Result<()> {
```

Comment:
Just for consistency, can we write these `crate::...` as `use` thingies at the top of the file?